onedal/linear_model/tests/test_linear_regression.py

```diff
@@ -0,0 +1,129 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import pytest
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+
+from onedal.linear_model import LinearRegression
+from onedal.tests.utils._device_selection import get_queues
+
+from sklearn.datasets import load_diabetes
+from sklearn.metrics import mean_squared_error
+from sklearn.model_selection import train_test_split
+
+
+@pytest.mark.parametrize('queue', get_queues())
+def test_diabetes(queue):
+    X, y = load_diabetes(return_X_y=True)
+    X_train, X_test, y_train, y_test = \
+        train_test_split(X, y,
+                         train_size=0.8, random_state=777)
+    model = LinearRegression(fit_intercept=True)
+    model.fit(X_train, y_train, queue=queue)
+    y_pred = model.predict(X_test, queue=queue)
+    assert mean_squared_error(y_test, y_pred) < 2396
+
+
+@pytest.mark.parametrize('queue', get_queues())
+def test_pickle(queue):
+    assert len(get_queues())
+    X, y = load_diabetes(return_X_y=True)
+    model = LinearRegression(fit_intercept=True)
+    model.fit(X, y, queue=queue)
+    expected = model.predict(X, queue=queue)
+
+    import pickle
+    dump = pickle.dumps(model)
+    model2 = pickle.loads(dump)

```

Comment:
Avoid usage of pickle or made serialization more secure in another way

Reply:
It is test. There is no need in extra security measures in this case.

Reply:
CodeFactor check should be greened. I guess some exception for it should be specified here

Reply:
I believe it will be fixed